ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/command/editor/page/text/VirtualFileImpl.java

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.command.editor.page.text;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+
+import org.eclipse.che.api.promises.client.Promise;
+import org.eclipse.che.api.promises.client.js.Promises;
+import org.eclipse.che.ide.api.resources.SyntheticFile;
+import org.eclipse.che.ide.api.resources.VirtualFile;
+import org.eclipse.che.ide.resource.Path;
+
+/**
+ * Copy of the {@link SyntheticFile} with ability to update it's content
+ * since {@link SyntheticFile} is read only.
+ *
+ * @see SyntheticFile
+ */
+class VirtualFileImpl implements VirtualFile {
```

Comment:
Not good name for this class, as i Understand this need for Command editor panel so propose use something like CommandData or CommandDescriptor.